### ISSUE TITLE
Audio dropouts every few minutes

### ISSUE THREAD

**Issue by skikirkwood** [mikebrady/shairport-sync#271]
I am running Shairport-sync from piCorePlayer 2.04.  After figuring out I need the "-c" option to have it use the hardware mixer for my IQAudio Pi-DAC+ it sounded great, but after a few minutes I get a dropout from a few second to a minute or more.  I turned on the -vvv option and this is what I see when a dropout occurs.  The Raspberry Pi 3 is not getting over 10% CPU load and the WiFi signal is very strong:

For long dropouts I get a "requesting resend of 1 packets.." followed by this:
Error -- ALSA device in incorrect state (4) for play.

For shorter dropouts I get this:

Player: packets out of sequence: expected: 53753, got: 53241, sync error: 0 frames.
Lost sync with source for 4 consecutive packets -- flushing and resyncing. Error: 180275.
Flush requested up to 1275498758. It seems as if 0 is special.
syncing to seqno 53417.
Open Mixer
Mixer device name is "hw:CARD=IQaudIODAC".
Mixer Control name is "Digital".
Setting volume db to -1673.750000.
requesting resend of 1 packets starting at 53461.
requesting resend of 1 packets starting at 53462.
requesting resend of 1 packets starting at 53463.
requesting resend of 1 packets starting at 53464.
requesting resend of 1 packets starting at 53465.
requesting resend of 1 packets starting at 53466.
.....

CSeq: 452.
DACP-ID: F89D2C38CF5DFB0B.
Active-Remote: 2330170117.
User-Agent: AirPlay/280.32.2.
Too-short retransmitted audio packet received in control port, ignored.
Too-short retransmitted audio packet received in control port, ignored.
Too-short retransmitted audio packet received in control port, ignored.
Too-short retransmitted audio packet received in control port, ignored.
Too-short retransmitted audio packet received in control port, ignored.
Too-short retransmitted audio packet received in control port, ignored.
.......

CSeq: 453.
DACP-ID: F89D2C38CF5DFB0B.
Active-Remote: 2330170117.
User-Agent: AirPlay/280.32.2.
Packet reception intervals: mean, standard deviation and max for the last 2,500 packets in microseconds: 8667.7, 65203.2, 3222038.0.
CSeq: 454.
DACP-ID: F89D2C38CF5DFB0B.
Active-Remote: 2330170117.
User-Agent: AirPlay/280.32.2.
CSeq: 455.
DACP-ID: F89D2C38CF5DFB0B.
Active-Remote: 2330170117.
User-Agent: AirPlay/280.32.2.

**Comment by skikirkwood** [comment]
BTW, I previously installed Shairport-sync on a Raspberry Pi B+ with Volumio 1.55 and had no dropout issues at all, it worked perfectly.

**Comment by mikebrady** [comment]
Thanks for the comprehensive report. Although the WiFi signal is strong and so on, the information points to some kind of network issue. First the message:

```
Player: packets out of sequence: expected: 53753, got: 53241, sync error: 0 frames.
```
is incorrect. (Notice the difference between the two numbers is 512, which is the total number of buffers available.) What it means is that all the buffers are empty, and the "next" buffer still has its old contents from 512 buffers in the past. This mistake is fixed in the current version – 2.8.2.

Now, the line that reads:
```
Packet reception intervals: mean, standard deviation and max for the last 2,500 packets in microseconds: 8667.7, 65203.2, 3222038.0.
```
seems to mean that there's a problem with the network. 
There are 352 audio frames in every packet, so, at 44,100 frames per second, the interval between packets should be 352/44,100 seconds, i.e. 7,981.86 microseconds. In your case, the average is 8,667.7 microseconds. The second figure standard deviation is a measure of the variation in arrival time intervals and, in my experience, should be down below 6,000, often well below. The maximum interval between packets is measured at 3,222,038, i.e. over three seconds! That is clearly way out of line, and should not exceed 200 milliseconds.

Here are some values from a Raspberry Pi 3 running an IQaudIO Digi-AMP over WiFi:

```
Apr 17 08:09:37 raspberrypi3 shairport-sync[5425]: Packet reception interval stats: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.2,     1432.0,    38512.0.
Apr 17 08:09:57 raspberrypi3 shairport-sync[5425]: Packet reception interval stats: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.1,     1571.9,    38178.0.
Apr 17 08:10:17 raspberrypi3 shairport-sync[5425]: Packet reception interval stats: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.4,     4629.6,    97958.0.
Apr 17 08:10:37 raspberrypi3 shairport-sync[5425]: Packet reception interval stats: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.3,     1558.2,    38484.0.
Apr 17 08:10:57 raspberrypi3 shairport-sync[5425]: Packet reception interval stats: mean, standard deviation and max for the last 2,500 packets in microseconds:     7983.1,     1281.0,    33114.0.
Apr 17 08:11:17 raspberrypi3 shairport-sync[5425]: Packet reception interval stats: mean, standard deviation and max for the last 2,500 packets in microseconds:     7981.9,     4625.3,    98142.0.
```

So, in summary, I'd be looking very closely at the network – has anything changed on it? Have you tried restarting the router, etc.?

**Comment by skikirkwood** [comment]
Thanks for taking a look at this Mike!  I did some more tests, and getting stats in the range of what you expect for the first five minutes or so - perfect playback.  Then the "requesting resend of 1 packets starting at... " starts again with the save error messages as described above.  Sounds like piCorePlayer is not picking up the latest release of Shairport-sync, so perhaps I'll just wait until an update and see if the problem goes away.  

Here's the stats from the latest run, right before the packet resend requests started:

Packet reception intervals: mean, standard deviation and max for the last 2,500 packets in microseconds:     7978.9,     1988.8,    45695.0.
    CSeq: 20.
    User-Agent: iTunes/12.3.3 (Macintosh; OS X 10.11.4).
    Client-Instance: 99D0CB48D25DC37D.
    DACP-ID: 99D0CB48D25DC37D.
    Active-Remote: 702814797.
Packet reception intervals: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.1,     1837.0,    25665.0.
Packet reception intervals: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.1,     1590.2,    32386.0.
    CSeq: 21.
    User-Agent: iTunes/12.3.3 (Macintosh; OS X 10.11.4).
    Client-Instance: 99D0CB48D25DC37D.
    DACP-ID: 99D0CB48D25DC37D.
    Active-Remote: 702814797.
Packet reception intervals: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.4,     2781.2,    44970.0.
    CSeq: 22.
    User-Agent: iTunes/12.3.3 (Macintosh; OS X 10.11.4).
    Client-Instance: 99D0CB48D25DC37D.
    DACP-ID: 99D0CB48D25DC37D.
    Active-Remote: 702814797.
Packet reception intervals: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.2,     2443.5,    38889.0.

Then I got this and the errors started:

Packet reception intervals: mean, standard deviation and max for the last 2,500 packets in microseconds:     7982.7,    11520.6,   154912.0.

**Comment by mikebrady** [comment]
Thanks for the update. It's almost as if a timer is timing out somewhere. Is there a power-save mode active in the WiFi system somewhere? What is piCorePlayer running on?

**Comment by skikirkwood** [comment]
You guessed it!  I am running piCorePlayer 2.04 (the latest release) on a Raspberry Pi 3 using the built-in WiFi.  After disabling the WiFi power management with the below command I had no more issues:

sudo iwconfig wlan0 power off

Working perfectly now!  Thanks for your help Mike!

Jim

**Comment by mikebrady** [comment]
That's great!
